### PR TITLE
Split all of util/FormField into separate modules that can be tree-shaken away

### PR DESCRIPTION
**Pull request by ianmcorvidae**
Got to here from noticing that react-select, and the `Autocomplete` component in this library, were in the bundles for sonora, even though we don't use those. Turns out, we do use `FormTextField`, though, and since tree-shaking is done at a file/module level, that meant we brought in all of `util/FormField`. And all its dependencies, which included the `Autocomplete` component, which itself had react-select as a dependency.

This is a big change, but it's also 100% copy-pasting things around, pretty much. I'm putting this up to see what people think of this sort of change (given the "pulling in unnecessary dependencies" problem, especially).

### CODE REVIEW

**Comment by sri2k1us** [comment]
@mian - I think we can safely delete `AutoComplete`  (and `react-select`) since Material-Ui has its with own `AutoComplete`. See https://material-ui.com/components/autocomplete/.  I have put together a clean-up sheet for ui-lib couple of weeks ago. See https://docs.google.com/spreadsheets/d/1NAZFhh2ZRqPExAWtlsf8FsMCp4d57b28EtC1paOXixw/edit?usp=sharing. Components marked `Delete` can be deleted from this lib.

**Comment by psarando** [comment]
> I think we can safely delete `AutoComplete` (and `react-select`) since Material-Ui has its with own `AutoComplete`. See https://material-ui.com/components/autocomplete/.

We will still need the `FormSearchField`, which is the field that is using `AutoComplete`, once we import Metadata Templates from the current UI:
https://github.com/cyverse-de/ui/blob/master/react-components/src/metadata/AstroThesaurusSearchField.js
https://github.com/cyverse-de/ui/blob/master/react-components/src/metadata/OntologyLookupServiceSearchField.js

So I don't think we can just delete `AutoComplete`, but we should replace it with https://material-ui.com/components/autocomplete, although I don't know how much refactoring that will entail.

**Comment by ianmcorvidae** [comment]
I'll definitely leave that sort of refactoring to whoever's adding it into Sonora -- but this PR if merged would mean that it's not in Sonora's code bundles until/unless we use it, in any case!

(also, I'm @ianmcorvidae here, not mian!)

File: src/util/forms/FormCheckboxTableCell.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/**
+ * @author psarando
+ */
+import Checkbox from "@material-ui/core/Checkbox";
+import TableCell from "@material-ui/core/TableCell";
+
+// Apparently only necessary for FastField, but maybe more correct for "vanilla" Field as well.
+const onCheckboxChange = (setFieldValue, fieldName) => (event, checked) => {
+    setFieldValue(fieldName, checked);
+};
```
Comment (psarando) [inline src/util/forms/FormCheckboxTableCell.js]: Why not import this from `FormCheckbox`?

File: src/util/forms/FormTextField.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import TextField from "@material-ui/core/TextField";
```
Comment (psarando) [inline src/util/forms/FormTextField.js:4]: Missing the `@author` doc in this file 😉

File: src/util/FormField.js [hunk 3]
```diff
@@ -1,266 +1,17 @@
...
+import FormCheckboxStringValue from "./forms/FormCheckboxStringValue";
+import FormCheckboxTableCell from "./forms/FormCheckboxTableCell";
+import FormIntegerField from "./forms/FormIntegerField";
+import FormMultilineTextField from "./forms/FormMultilineTextField";
+import FormNumberField from "./forms/FormNumberField";
+import FormSearchField from "./forms/FormSearchField";
+import FormSelectField from "./forms/FormSelectField";
+import FormTextField from "./forms/FormTextField";
+import FormTimestampField from "./forms/FormTimestampField";
+import getFormError from "./forms/getFormError";
```
Comment (psarando) [inline src/util/FormField.js]: I think these can be collapsed with the export below, by using statements like these:
```javascript
export { default as FormCheckbox } from "./forms/FormCheckbox";
export { default as FormCheckboxStringValue } from "./forms/FormCheckboxStringValue";
export { default as FormCheckboxTableCell } from "./forms/FormCheckboxTableCell";
```
...etc.

**Review by psarando — APPROVED** [review]
LGTM 👍

File: src/util/forms/FormCheckboxTableCell.js (hunk 1 above)
Comment (ianmcorvidae) [reply src/util/forms/FormCheckboxTableCell.js]: Oh, whoops. Probably could do that (if I have FormCheckbox export it, anyway)

File: src/util/FormField.js (hunk 3 above)
Comment (ianmcorvidae) [reply src/util/FormField.js]: Ah, good call. I'll double-check that does what's expected. It's a lot cleaner-looking.

File: src/util/forms/FormTextField.js (hunk 2 above)
Comment (ianmcorvidae) [reply src/util/forms/FormTextField.js:4]: ha, whoops, this is the one I did first. I'll add you back in :wink:

**Review by ash3rz — APPROVED** [review]
I didn't catch anything that wasn't already mentioned :shipit:

**Comment by ianmcorvidae** [comment]
Thanks for the reviews!

**Comment by sri2k1us** [comment]
> > I think we can safely delete `AutoComplete` (and `react-select`) since Material-Ui has its with own `AutoComplete`. See https://material-ui.com/components/autocomplete/.
> 
> We will still need the `FormSearchField`, which is the field that is using `AutoComplete`, once we import Metadata Templates from the current UI:
> https://github.com/cyverse-de/ui/blob/master/react-components/src/metadata/AstroThesaurusSearchField.js
> https://github.com/cyverse-de/ui/blob/master/react-components/src/metadata/OntologyLookupServiceSearchField.js
> 
> So I don't think we can just delete `AutoComplete`, but we should replace it with https://material-ui.com/components/autocomplete, although I don't know how much refactoring that will entail.

My reasoning behind delete is that:
- This is not currently used in Sonora
- This will need to reimplemented in With Material-UI autocomplete.
- Deleting here will not impact production DE
- `react-select` blots package size (am not sure if thats fixed with @ianmcorvidae update)

But am OK with leaving it as such if you feel strongly about it.

**Comment by sri2k1us** [comment]
Welp. I just read @ianmcorvidae comments on top of this page. I guess leaving it here or deleting would not really change lib size. 👍

**Pull request merged**